modules/optimization/src/optimizationreporters/OptimizationReporterBase.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptimizationReporterBase.h"
+
+InputParameters
+OptimizationReporterBase::validParams()
+{
+  InputParameters params = OptimizationData::validParams();
+  params.registerBase("OptimizationReporterBase");
```

Comment:
what does registerBase do?  I've never seen it.  I looked at InputParamerts.C and I don't understand what its doing.

Reply:
It makes sure that not any action can add this type of object. I.e. only the `OptimizationReporter` syntax can build the derived objects, as apposed to the `Reporters` syntax.